muxtools/subtitle/font.py

```diff
@@ -143,8 +192,115 @@ def clean_name(fontname: str) -> str:
 
             if not outpath.exists():
                 shutil.copy(fontpath, outpath)
+            
+            if subset_fonts:
+                # We always want to include the used characters, and any additional glyphs specified by the user
+                characters_set = usage_data.characters_used
+                characters_set.update(subset_additional_glyphs_parsed)
+
+                if not subset_aggressive:
+                    # We also want to add a general subset of common characters to avoid issues when reusing fonts, if we aren't doing aggressive subsetting
+                    characters_set.update(COMMON_UNICODE_CHARS)
+
+                if not subset_ignore_fonts_with_no_usage and not characters_set:
+                    # If theres no characters used, and we aren't ignoring fonts with no usage, add common characters
+                    warn(f"No characters used in font '{fontname}'. Defaulting to common subset.", collect_fonts)
+                    characters_set.update(COMMON_UNICODE_CHARS)
+
+                if not characters_set:
+                    warn(f"No characters used in font '{fontname}'. Skipping subsetting...", collect_fonts)
+                
+                else:
```

Comment:
for cleaner control flow, consider simply `continue`ing from the above `if`? the blank line between `if` and `else` makes this easy to misunderstand as unrelated blocks of code